static/app/components/globalDrawer/components.tsx

```diff
@@ -28,30 +40,182 @@ interface DrawerPanelProps {
+      e.preventDefault();
+
+      const handle = resizeHandleRef.current;
+      const panel = panelRef.current;
+      if (!handle || !panel) {
+        return;
+      }
+
+      // Mark as resizing
+      handle.setAttribute('data-resizing', 'true');
+      panel.setAttribute('data-resizing', '');
+      initialMousePositionRef.current = e.clientX;
+
+      const viewportWidth = typeof window === 'undefined' ? 1000 : window.innerWidth;
+
+      const handleMouseMove = (moveEvent: MouseEvent) => {
+        moveEvent.preventDefault();
+
+        if (rafIdRef.current !== null) {
+          window.cancelAnimationFrame(rafIdRef.current);
+        }
+
+        rafIdRef.current = window.requestAnimationFrame(() => {
+          if (!panel || !handle || initialMousePositionRef.current === null) {
+            return;
+          }
+
+          const newWidthPercent =
+            ((viewportWidth - moveEvent.clientX) / viewportWidth) * 100;
+
+          panel.style.setProperty('--drawer-width', `${newWidthPercent}%`);
+
+          // Update handle attributes for cursor styles
+          handle.setAttribute(
+            'data-at-min-width',
+            (newWidthPercent <= MIN_WIDTH_PERCENT).toString()
+          );
+          handle.setAttribute(
+            'data-at-max-width',
+            (Math.abs(newWidthPercent - MAX_WIDTH_PERCENT) < 1).toString()
+          );
+        });
+      };
+
+      const handleMouseUp = () => {
+        if (rafIdRef.current !== null) {
+          window.cancelAnimationFrame(rafIdRef.current);
+          rafIdRef.current = null;
+        }
+
+        if (handle) {
+          handle.removeAttribute('data-resizing');
+        }
+
+        if (panel) {
+          panel.removeAttribute('data-resizing');
+          // Get the computed width considering min/max constraints and save to localStorage
+          const computedStyle = window.getComputedStyle(panel);
+          const widthValue = (parseFloat(computedStyle.width) / viewportWidth) * 100;
+          setPersistedWidthPercent(widthValue);
+        }
+
+        document.removeEventListener('mousemove', handleMouseMove);
+        document.removeEventListener('mouseup', handleMouseUp);
+      };
+
+      document.addEventListener('mousemove', handleMouseMove);
+      document.addEventListener('mouseup', handleMouseUp);
+    },
+    [setPersistedWidthPercent]
+  );
+
+  useLayoutEffect(() => {
+    return () => {
+      if (rafIdRef.current !== null) {
+        window.cancelAnimationFrame(rafIdRef.current);
+      }
+    };
+  }, []);
+
   return (
     <DrawerContainer>
       <DrawerSlidePanel
         ariaLabel={ariaLabel}
         slidePosition="right"
         collapsed={false}
-        ref={ref}
+        ref={node => {
+          panelRef.current = node;
+          if (typeof ref === 'function') {
+            ref(node);
+          } else if (ref) {
+            (ref as React.RefObject<HTMLDivElement | null>).current = node;
+          }
+        }}
```

Comment:
I think you just want to use `import {mergeRefs} from '@react-aria/utils';` for this